src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php

```diff
@@ -27,11 +28,20 @@
  */
 class DoctrineExtractor implements PropertyListExtractorInterface, PropertyTypeExtractorInterface
 {
+    private $entityManager;
     private $classMetadataFactory;
 
-    public function __construct(ClassMetadataFactory $classMetadataFactory)
+    /**
+     * @param EntityManagerInterface $entityManager
+     */
+    public function __construct($entityManager)
     {
-        $this->classMetadataFactory = $classMetadataFactory;
+        if ($entityManager instanceof EntityManagerInterface) {
+            $this->entityManager = $entityManager;
+        } else {
```

Comment:
`elseif ($entityManager instanceof ClassMetadataFactory)` and then we should add an `else` block that throws an `InvalidArgumentException` for everything else

Reply:
We don't check for invalid types when documented in the DockBlock, do we?

Reply:
For other places where we deprecated arguments and removed the type hint we added them.